.chloggen/k8sattributes-remove-deprecated-association-fields.yaml

```diff
@@ -0,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: k8sattributes
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Remove support of deprecated `pod_association` fields
```

Comment:
nit: Add subtext explaining how to migrate from this to the new format (same as your PR description)

Reply:
I don't think we should put it in the changelog for the deprecation fields removal. They were deprecated half a year ago. I think such subtext should be part of the changelog items where the fields are deprecated, not when they are removed. WDYT?

Reply:
I think it's important to point people to the new fields, since for some people this will be the first time they see this. A subtext saying *Fields are now nested under the `pod_association::sources` subsection* seems like enough info to me to answer that. Does that make sense?

I am going to approve since I don't want to block on a nit, but please consider this :)

Reply:
Sure, updated :)